docs/overview/Lifecycle.md

```diff
@@ -0,0 +1,335 @@
+serves as our entry point.
+[You can read more about the module map and how it is structured.](../api/server/Module-Map-Schema.md)
+
+One App uses the environment variable `HOLOCRON_MODULE_MAP_URL` to configure from where
+the module map is loaded (as `JSON`). In `development`, there is a local CDN ran alongside One App
+to serve local modules and the module map - please note that when running One App directly
+(repository or Docker image) or using `one-app-runner` will automatically configure this for us.
+When in `production` mode, this variable is required.
+
+>  * [`HOLOCRON_MODULE_MAP_URL`](../api/server/environment-variables#holocron_module_map_url) ⚠️
+
+You can further configure the behavior of Holocron with the Server Settings below:
+
+>  * [`HOLOCRON_SERVER_MAX_MODULES_RETRY`](../api/server/environment-variables#holocron_server_max_modules_retry)
+>  * [`HOLOCRON_SERVER_MAX_SIM_MODULES_FETCH`](../api/server/environment-variables#holocron_server_max_sim_modules_fetch)
+>  * [`ONE_MAP_POLLING_MAX`](../api/server/environment-variables#one_map_polling_max)
+>  * [`ONE_MAP_POLLING_MIN`](../api/server/environment-variables#one_map_polling_min)
+
+One App will first fetch the Holocron module map using the provided URL
+and proceeds to load every module defined in the module map into memory.
+If a module failed to load for any reason or when validating, One App will
+log the error and recover gracefully, adding the module to a `block` list until
+updated. Holocron modules are reloaded whenever the module map is polled and does
+not match what is on the server.
+
+After the modules are loaded and validated, One App can start accepting requests.
+Once One App becomes operational, it begins to periodically poll the module map
+URL supplied by the user to observe for any changes. If a Holocron module is added,
+updated or removed from the module map, changed modules will be re-installed into
+memory.
+
+## Runtime App Configuration
+
+As we discussed in the [Getting Started Guide](../Getting-Started.md), each
+Holocron module can supply an [`appConfig`](../api/modules/App-Configuration.md)
+that is executed on and configures the server.
+
+An example `Module.appConfig`:
+
+```jsx
+import React from 'react';
+
+export default function RootModule({ children }) {
+  return children;
+}
+
+if (!global.BROWSER) {
+  RootModule.appConfig = {
+    csp: 'default-src \'self\'',
+  };
+}
+```
+
+We can use this additional configuration interface to further tweak
+One App to our needs. The only requirement is that the root module
+needs to be configured with a valid CSP.
+
+## Server Side Rendering
+
+When One App has loaded all of our modules and started the `express` server,
+the app becomes ready to render our Holocron modules. There are a few ways that
+One App can render our Holocron modules:
+
+- default render path, using the request URL to render the modules that match
+- partial renders, renders only the selected Holocron module in isolation
```

Comment:
partials are more than just modules in isolation, it is a different react render method https://reactjs.org/docs/react-dom-server.html#rendertostaticmarkup maybe we should add this note in the expanded explanation of partials below

Reply:
Valid points, I will correct this.

Reply:
I have updated this (401b543)